src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/JsonSchemaServiceTest.kt

```diff
@@ -0,0 +1,108 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.unit.service
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import io.mockk.every
+import io.mockk.mockk
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.ApplicationEntityFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.ApplicationSchemaEntityFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.ProbationOfficerEntityFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ApplicationEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ApplicationRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ApplicationSchemaRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.JsonSchemaService
+import java.util.UUID
+
+class JsonSchemaServiceTest {
```

Comment:
Was there a test for `validate` before? Would be good to have both

Reply:
So this isn't actually called by anything other than `attemptSchemaUpgrade` - had meant to make it private hence the lack of tests.  Have made private now 👍

Reply:
Ah, gotcha. I imagine it'll be used elsewhere at some point though right?

Reply:
Yeah will need it during creation, will pop the unit tests in for it then